core/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -94,8 +96,13 @@ public final class OpenTelemetryRumBuilder {
 
     private Resource resource;
 
-    @Nullable private ServiceManager serviceManager;
-    @Nullable private ExportScheduleHandler exportScheduleHandler;
+    private final Object lock = new Object();
+
+    // Writes guarded by "lock"
+    @Nullable private volatile ServiceManager serviceManager;
+
+    // Writes guarded by "lock"
+    @Nullable private volatile ExportScheduleHandler exportScheduleHandler;
```

Comment:
It was mostly due to an uncertainty if they were allowed to be changed after calling `build`. Happy to remove this and perhaps add a check to ensure you're not calling these after calling `build`.

Reply:
Got it, yeah none of the setters should be called after calling `build()`, though I see your point that it's not impossible to do so right now as we don't have validations in place for it, so yeah I think we should add checks to prevent it from happening. We could add those validations here or as part of a follow-up PR too, so no worries if wouldn't like to address it now 👍 though I think we should add those validations for sure at some point.

Reply:
Updated and checks added